pkg/alertmanager/alertmanager.go

```diff
@@ -173,6 +176,11 @@ func clusterWait(p *cluster.Peer, timeout time.Duration) func() time.Duration {
 
 // ApplyConfig applies a new configuration to an Alertmanager.
 func (am *Alertmanager) ApplyConfig(userID string, conf *config.Config) error {
+	// Ensure the alertmanager is set to active
+	am.activeMtx.Lock()
+	am.active = true
+	am.activeMtx.Unlock()
```

Comment:
Shouldn't this done at the end of the method, once we're sure we'll not return error and everything has been setup correctly?

Reply:
Yes, I think that would be better.